pkg/proxy/proxy.go

```diff
@@ -172,45 +186,61 @@ func (p *Proxy) CreateOrUpdateRedirect(l4 *policy.L4Filter, id string, localEndp
 
 	scopedLog := log.WithField(fieldProxyRedirectID, id)
 
-	if r, ok := p.redirects[id]; ok {
-		r.mutex.Lock()
-		defer r.mutex.Unlock()
+	revertFuncs := make([]RevertFunc, 0, 2)
+	revertFunc = func() error {
+		for i := len(revertFuncs) - 1; i >= 0; i-- {
+			if err := revertFuncs[i](); err != nil {
```

Comment:
Shouldn't we try to run all revert funcs if possible? And then return a list of errors? I guess we will have inconsistent state either way, though :/

Reply:
There are typically dependencies between the revert functions, so that's not possible.